templates/kotlin/src/main/java/io/appwrite/Client.kt.twig

```diff
@@ -0,0 +1,221 @@
+        selfSigned = status
+
+        val cookieManager = CookieManager().apply {
+            setCookiePolicy(CookiePolicy.ACCEPT_ALL)
+        }
+        
+        val builder = OkHttpClient()
+            .newBuilder()
+            .cookieJar(JavaNetCookieJar(cookieManager))
+            .addInterceptor(HttpLoggingInterceptor().apply { setLevel(HttpLoggingInterceptor.Level.BODY) })
+
+        if (!selfSigned) {
+            http = builder.build()
+            return this
+        }
+
+        try {
+            // Create a trust manager that does not validate certificate chains
+            val trustAllCerts = arrayOf<TrustManager>(
+                object : X509TrustManager {
+                    @SuppressLint("TrustAllX509TrustManager")
+                    override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {
+                    }
+                    @SuppressLint("TrustAllX509TrustManager")
+                    override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {
+                    }
+                    override fun getAcceptedIssuers(): Array<X509Certificate> {
+                        return arrayOf()
+                    }
+                }
+            )
+            // Install the all-trusting trust manager
+            val sslContext = SSLContext.getInstance("SSL")
+            sslContext.init(null, trustAllCerts, SecureRandom())
+
+            // Create an ssl socket factory with our all-trusting manager
+            val sslSocketFactory: SSLSocketFactory = sslContext.socketFactory
+            builder.sslSocketFactory(sslSocketFactory, trustAllCerts[0] as X509TrustManager)
+            builder.hostnameVerifier(HostnameVerifier { _, _ -> true })
+
+            http = builder.build()
+        } catch (e: Exception) {
+            throw RuntimeException(e)
+        }
+
+        return this
+    }
+
+    fun setEndpoint(endPoint: String): {{ spec.title | caseUcfirst }}Client {
+        this.endPoint = endPoint
+        return this
+    }
+
+    fun addHeader(key: String, value: String): {{ spec.title | caseUcfirst }}Client {
+        headers[key] = value
+        return this
+    }
+
+    @Throws({{ spec.title | caseUcfirst }}Exception::class)
+    suspend fun call(
+        method: String, 
+        path: String, 
+        headers:  Map<String, String>, 
+        params: Map<String, Any?>
+    ): Response {
+        val requestHeaders = this.headers.toHeaders().newBuilder()
+            .addAll(headers.toHeaders())
+            .build()
+
+        val httpBuilder = (endPoint + path).toHttpUrl().newBuilder()
+
+        if ("GET" == method) {
+            params.forEach {
+                if (it.value is List<*>) {
+                    httpBuilder.addQueryParameter(it.key + "[]", it.value.toString())
+                } else {
+                    httpBuilder.addQueryParameter(it.key, it.value.toString())
```

Comment:
@abnegate  Here's another thing I noticed in case of `DatabaseService.listDocuments()`
When we don't pass filters, the default argument is `null`

```kotlin
suspend fun listDocuments(
		collectionId: String,
		filters: List<Any>? = null,
		limit: Int = 25,
		offset: Int = 0,
		orderField: String = "",
		orderType: OrderType = OrderType.ASC,
		orderCast: String = "string",
		search: String = ""
	): Response
` ``
When we make the `GET` call using `client.call()`,  filters stays nulll and results in a URL like this 
```
https://demo.appwrite.io/v1/database/collections/60632e9bb9631/documents?filters=null&limit=25&offset=0&orderField=&orderType=ASC&orderCast=string&search=
```

`filters=null` causes a Server Side 500 error. 
 
Let's replace it so that only the `non-null params` in the hashmap are used in the query parameters. If there is a better way to do it, I'm welcome to that as well 😄  

```kotlin
 it.value?.let { it2 ->
                    if (it2 is List<*>) {
                        httpBuilder.addQueryParameter(it.key + "[]", it.value.toString())
                    } else {
                        httpBuilder.addQueryParameter(it.key, it.value.toString())
                    }
              }
```